client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,380 @@
+    transition: { default: 150, type: Number },
+    selected: { default: '', type: String },
+    syncURLHash: { default: false, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  /** ID of the currently selected tab. */
+  declare selectedValue: string;
+  /** If true, animation will run when a new tab is selected. */
+  declare readonly animateValue: boolean;
+  /** Tab elements, with role='tab', allowing switching between tabs. */
+  declare readonly labelTargets: HTMLAnchorElement[];
+  /** Tab content panels, with role='tabpanel', showing the content for each tab. */
+  declare readonly panelTargets: HTMLElement[];
+  /** Other elements within the controller's scope that may trigger a specific tab. */
+  declare readonly triggerTargets: HTMLAnchorElement[];
+  /** Classes to set on the tab panel content when selected. */
+  declare readonly selectedClasses: string[];
+  /** If true, the selected tab will sync with the URL hash. */
+  declare readonly syncURLHashValue: boolean;
+  /** The time in milliseconds for the tab content to transition in and out. */
+  declare readonly transitionValue: number;
+
+  connect() {
+    this.validate();
+
+    debounce(() => {
+      window.scrollTo({ top: 0, behavior: 'smooth' });
+    }, this.transitionValue * 2);
+
+    this.setAriaControls(this.labelTargets);
+    this.setTabLabelIndex();
+
+    const activeTab = this.labelTargets.find(
+      (button) =>
+        button.getAttribute('aria-selected') === 'true' ||
+        button.getAttribute('aria-controls') === this.selectedValue,
+    );
+
+    if (this.selectedClasses.length && activeTab) {
+      activeTab.setAttribute('aria-selected', 'true');
+      activeTab.removeAttribute('tabindex');
+    }
+
+    this.panelTargets.forEach((tab) => {
+      // eslint-disable-next-line no-param-reassign
+      tab.hidden = true;
+    });
+
+    if (window.location.hash && !this.syncURLHashValue) {
+      this.setTabByURLHash();
+    } else if (activeTab) {
+      this.selectedValue = activeTab.getAttribute('aria-controls') as string;
+    } else {
+      this.selectFirstTab();
+    }
+
+    this.setAriaControls(this.triggerTargets);
+  }
+
+  selectedValueChanged(currentValue: string, previousValue: string) {
+    if (previousValue) {
+      this.hideTabContent(previousValue);
+    }
+
+    const tab = this.getTabLabelByHref(currentValue);
+    if (tab) {
+      tab.setAttribute('aria-selected', 'true');
+      tab.removeAttribute('tabindex');
+    }
+
+    const tabContent = this.getTabPanelByHref(currentValue);
+
+    if (tabContent) {
+      if (this.animateValue) {
+        this.animateIn(tabContent);
+      } else {
+        tabContent.hidden = false;
+      }
+
+      this.dispatch('switch', {
+        detail: { tab: tab?.getAttribute('href')?.replace('#', '') },
+        target: tab,
+      });
+
+      this.dispatch('selected', {
+        cancelable: false,
+        detail: { selected: currentValue },
+      });
```

Comment:
Let's merge these events into one, sorry for confusion before on the events but I think we can simplify this.

```suggestion
      this.dispatch('selected', {
        cancelable: false,
        detail: { selected: currentValue },
        target: tab,
      });
```

Remember to also update the code that listens to this event in `client/src/entrypoints/admin/comments.js`

```
      tabNavElement.addEventListener('switch', (e) => {
        commentApp.setCurrentTab(e.detail.tab);
      });
```

Should become

```js
      tabNavElement.addEventListener('w-tabs:selected', ({ detail: {selected}}) => {
        commentApp.setCurrentTab(selected);
      });
```

Be sure to manually check that works though.